commons/zenoh-buffers/src/zslice.rs

```diff
@@ -122,7 +122,7 @@ impl ZSlice {
     }
 
     pub fn empty() -> Self {
-        unsafe { ZSlice::new_unchecked(Arc::new([]), 0, 0) }
+        Self::new(Arc::new([]), 0, 0).unwrap()
```

Comment:
`unwrap()` add runtime code to handle panics. Here it seems unnecessary to me.

Reply:
I adds it only if the check is not elided, see https://godbolt.org/z/5G1795dn1